### ISSUE TITLE
midgard_connection::open() and open_config() do not create classes

### ISSUE THREAD

**Issue by indeyets** [midgardproject/midgard-php5#9]
In case, when connection is established manually using midgard_connection::open() or midgard_connection::open_config() methods and schema-directory in config-file is different from the schema-directory originally set in midgard, mgdschema classes (descendants of midgard_object) are not initialized.

Fixing this, would require to move initialization of these classes from PHP_MINIT_FUNCTION(midgard2_object) to some other function and call that function after new config is loaded

**Comment by piotras** [comment]
It's also related to classes' type. If it's user type, it should be freed at request end. If it's internal one, it should be initialized during MINIT as it guaranteed to register classes in main process before any new child is forked.

**Comment by indeyets** [comment]
@piotras this one is strictly about "schema" classes, which, in case of `midgard_connection::open()` or `midgard_connection::open_config()` have to be initialized/deinitialized during request lifetime.

**Comment by piotras** [comment]
Yes, this is what I meant. It is unsafe to register internal class in child process, and doesn't make sense to register all mgdschema classes per every request. Unless, there are special routines which take cli and server context into account. But this one might be a bit complicated.

**Comment by bergie** [comment]
This is a major annoyance when using Ratatoskr outside of Apache (with AiP, for instance). For now the workaround has been to use the `MIDGARD_ENV_GLOBAL_SHAREDIR` environment variable before starting the PHP process that uses Midgard. But that is clunky when everything else is handled via `midgard_config` object or file.

The current state causes weird workarounds, like:
https://github.com/piotras/phpcr-midgard2/blob/master/src/Midgard/PHPCR/RepositoryFactory.php#L25
https://github.com/piotras/phpcr-midgard2/blob/master/api-test/bootstrap.php#L22
